utils.h

```diff
@@ -65,5 +65,23 @@
         }\
     } while (0)
 
+
+/* Workaround missing __rdtsc in Clang < 3.5 (or Clang < 6.0 on Xcode) */
+#if defined(__x86_64__) || defined(__i386__)
+#if defined __clang__
+#if (!defined(__apple_build_version__) &&\
+        ((__clang_major__ < 3) || ((__clang_major__ == 3) && (__clang_minor__ < 5))))\
+    || (defined(__apple_build_version__) && (__clang_major__ >= 6))
+static inline uint64_t __rdtsc() {
```

Comment:
GCC and clang >= 3.5 can do a better job at optimising with builtins i.e. they resolve to `__builtin_ia32_rdtsc()` rather than an inline assembly block.